Tasks/Gradle/sonarqube-common/sonarqube-common.ts

```diff
@@ -205,16 +212,47 @@ function getOrCreateSonarQubeStagingDirectory(): string {
     return sqStagingDir;
 }
 
+// Creates a string containing Markdown of the SonarQube quality gate status for this project.
+function createQualityGateStatusSection(taskReport: TaskReport): Q.Promise<string> {
+    if (!tl.getBoolInput('sqAnalysisWaitForAnalysis')) {
+        // Do not create a quality gate status section if not waiting for the server to analyse the build
+        console.log(tl.loc('sqCommon_NotWaitingForAnalysis'));
+        return Q.when<string>(null);
+    }
+
+    // Wait for analysis to finish so that we can get analysis information from the SonarQube server
+    var timeout:number = 60;
+    var delay:number = 1;
+
+    console.log(tl.loc('sqCommon_WaitingForAnalysis'));
+    return sqRest.waitForAnalysisCompletion(taskReport, timeout, delay)
+        .then((analysisCompleted:boolean) => {
+            if (!analysisCompleted) {
+                tl.debug('Did not receive a success response before the timeout expired.');
+                return Q.reject(tl.loc('sqAnalysis_AnalysisTimeout', timeout));
+            }
+
+            // If analysis completed, fetch the analysis ID to check quality gate status
+            return sqRest.getAnalysisId(taskReport)
```

Comment:
Consider using a linter, like TsLint that will suggest minor corrections like using ;